contentcuration/contentcuration/utils/recommendations.py

```diff
@@ -61,67 +68,200 @@ def create_backend(self) -> Backend:
 
 class RecommendationsAdapter(Adapter):
 
-    def generate_embedding(self, text) -> EmbeddingsResponse:
-        request = EmbeddingsRequest()
-        return self.backend.make_request(request)
-
-    def embedding_exists(self, embedding) -> bool:
-        # Need to implement the logic to check if the embeddigns exist
-        # Return True if the embedding exists, or False otherwise
-        return True
-
-    def cache_embeddings(self, embeddings_list) -> bool:
-        for embedding in embeddings_list:
-            try:
-                # Attempt to cache the embedding
-                # Write the caching logic
-                # A conrner case to look at here is if one of the embedding fails to get cached
-                # we need to handel it so that only the once that were not succesfull
-                # are attempted to cache again
-                pass
-            except Exception as e:
-                print(e)
-                return False
-        return True
-
-    def get_recommendations(self, embedding) -> RecommendationsResponse:
-        request = RecommendationsRequest(embedding)
-        return self.backend.make_request(request)
-
-    def embed_topics(self, topics: Dict[str, Any]) -> EmbeddingsResponse:
+    def generate_embeddings(self, request: EmbeddingsRequest,
+                            cache: bool = True) -> EmbeddingsResponse:
+        """
+        Generates recommendations for the given request.
 
+        :param request: The request for which to generate embeddings.
+        :param cache: Whether to cache the embeddings request.
+        :return: The response containing the recommendations.
+        :rtype: EmbeddingsResponse
+        """
         if not self.backend.connect():
             raise errors.ConnectionError("Connection to the backend failed")
 
+        cached_response = self.response_exists(request)
```

Comment:
There's a bit of a mixup here. We don't need to cache the embeddings-- we cache the response from the recommendations API returning recommended nodes. Those need to be 2 separate APIs, so having the caching logic tied to embeddings on the Studio side is mixing up the two.

Reply:
Cleaned this up by moving out the caching logic